From 31767694c4e459cd4ee8b5bc246054b01cf27f85 Mon Sep 17 00:00:00 2001
From: yerriswamy <yerriswamy.kt@intel.com>
Date: Thu, 2 Feb 2023 21:26:00 +0530
Subject: [PATCH] Resolved compilation issue for media driver

Changes include:
- Added Android flag to disable error mode similar to
Linux

Change-Id: Iad43d1d036783dabca1b18c2bade9f882b56a8a0
Tracked-On: OAM-105683
Signed-off-by: Ratnesh Kumar Rai <ratnesh.kumar.rai@intel.com>
Signed-off-by: yerriswamy <yerriswamy.kt@intel.com>
---
 cmrtlib/linux/hardware/cm_device_os.cpp       |   6 +-
 media_driver/Android.mk                       | 172 +++++++++++++++++-
 .../common/vp/kdll/hal_kerneldll_next.c       |   5 +-
 .../ddi/media_libva_caps_mtl_base.cpp         |   1 +
 .../os/osservice/mos_utilities_specific.cpp   |   2 +
 5 files changed, 177 insertions(+), 9 deletions(-)

diff --git a/cmrtlib/linux/hardware/cm_device_os.cpp b/cmrtlib/linux/hardware/cm_device_os.cpp
index 36b98c13a..e21292283 100644
--- a/cmrtlib/linux/hardware/cm_device_os.cpp
+++ b/cmrtlib/linux/hardware/cm_device_os.cpp
@@ -495,8 +495,10 @@ CmDevice_RT::CmDevice_RT(
     m_gtpinBufferUP1(nullptr),
     m_gtpinBufferUP2(nullptr),
     m_createOption(createOption),
-    m_driverStoreEnabled(0),
-    m_driFileDescriptor(0)
+    m_driverStoreEnabled(0)
+#ifndef ANDROID
+    ,m_driFileDescriptor(0)
+#endif
 {
 
     // New Surface Manager
diff --git a/media_driver/Android.mk b/media_driver/Android.mk
index d1b44d39e..8d36c4dff 100644
--- a/media_driver/Android.mk
+++ b/media_driver/Android.mk
@@ -425,7 +425,6 @@ LOCAL_SRC_FILES := \
     ../media_softlet/agnostic/common/os/mos_user_setting.cpp \
     ../media_softlet/agnostic/common/os/mos_util_debug.cpp \
     ../media_softlet/agnostic/common/os/mos_utilities_next.cpp \
-    ../media_softlet/agnostic/common/os/private/mos_utilities_usersetting.cpp \
     ../media_softlet/agnostic/common/renderhal/renderhal.cpp \
     ../media_softlet/agnostic/common/renderhal/renderhal_platform_interface_next.cpp \
     ../media_softlet/agnostic/common/shared/bufferMgr/media_allocator.cpp \
@@ -587,10 +586,6 @@ LOCAL_SRC_FILES := \
     ../media_softlet/linux/common/os/mos_vma.c \
     ../media_softlet/linux/common/os/osservice/mos_util_debug_specific.cpp \
     ../media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp \
-    ../media_softlet/linux/common/os/private/mos_decompression.cpp \
-    ../media_softlet/linux/common/os/private/mos_mediacopy.cpp \
-    ../media_softlet/linux/common/os/private/mos_os_specific.cpp \
-    ../media_softlet/linux/common/os/private/mos_utilities_specific_usersetting.cpp \
     ../media_softlet/linux/common/shared/hal_oca_interface_next.cpp \
     ../media_softlet/linux/common/shared/skuwa_dumper_specific.c \
     ../media_softlet/linux/common/shared/user_setting/media_user_setting_configure_specific.cpp \
@@ -1872,6 +1867,173 @@ LOCAL_C_INCLUDES  = \
     $(LOCAL_PATH)/../media_softlet/linux/xe_lpm_plus_r0/decode/vp8/ddi \
     $(LOCAL_PATH)/../media_softlet/linux/xe_lpm_plus_r0/decode/vp9/ddi \
     $(LOCAL_PATH)/../media_softlet/media_interface/media_interfaces_mtl \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/shared \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/cp \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/vp/hal/utils/hal_ddi_share \
+    $(LOCAL_PATH)/../media_softlet/linux/common/vp/ddi \
+    $(LOCAL_PATH)/../media_softlet/linux/common/ddi \
+    $(LOCAL_PATH)/../media_softlet/linux/common/codec/ddi/shared \
+    $(LOCAL_PATH)/../media_softlet/linux/common/codec/ddi/enc \
+    $(LOCAL_PATH)/../media_softlet/linux/common/codec/ddi/dec \
+    $(LOCAL_PATH)/../media_softlet/linux/common/cp \
+    $(LOCAL_PATH)/../media_softlet/linux/common/cp/ddi \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/shared/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/shared/profiler \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/shared/media_sfc_interface \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/shared/mediacopy \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/shared/statusreport \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/shared/mmc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/shared/bufferMgr \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/shared/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/shared/task \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/shared/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/avc/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/avc/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/avc/features/roi \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/avc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/avc/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/avc/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/shared \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/shared/bitstreamWriter \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/jpeg/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/shared/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/shared/mmc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/shared/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/shared/scalability \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/shared/bufferMgr \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/shared/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/hevc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/av1 \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/av1/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/av1/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/av1/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/shared \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/hevc/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/hevc/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/hevc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/hevc/features/roi \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/shared/statusreport \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/vp9/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/vp9/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/vp9/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/jpeg/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/enc/jpeg/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/shared \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/avc/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/shared/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/shared/hucItf \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/shared/statusreport \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/shared/mmc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/shared/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/mpeg2/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/mpeg2/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/mpeg2/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/mpeg2/mmc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/vp9/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/vp9/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/vp9/mmc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/vp9/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/vp9/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/vp9/scalability \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/vp8/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/vp8/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/vp8/mmc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/vp8/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/hevc/scalability \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/jpeg/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/jpeg/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/jpeg/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/avc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/shared/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/hevc/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/hevc/scalability \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/shared/scalability \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/shared/bufferMgr \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/hevc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/hevc/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/hevc/mmc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/avc/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/av1/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/av1/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/av1/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/codec/hal/dec/jpeg/bitstream \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/jpeg/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/vp/hal/feature_manager \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/vp/hal/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/shared/mediacopy \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/avc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/avc/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/av1/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/av1/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/vp/hal/feature_manager \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/vp/hal/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/shared/mediaDecompress \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/vp9/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/vp9/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/hevc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/vp9/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/av1/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/hevc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/mpeg2/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/avc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/vp8/mmc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/vp8/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/av1/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/av1/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/mpeg2/mmc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/av1/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/avc/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/av1/features/preenc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/hevc/mmc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/hevc/features/preenc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/shared/common \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/hw/vdbox \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/shared \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/enc/hevc/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/vp9/mmc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/vp8/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/hevc/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/mpeg2/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/vp9/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/shared \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/jpeg/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/codec/hal/dec/jpeg/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/hw \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/dec/avc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/dec/hevc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/vp/hal/feature_manager \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/dec/av1/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/dec/mpeg2 \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/dec/vp9/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/enc/hevc \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/dec/vp8/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/enc/av1/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/enc/vp9/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/vp/hal/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/enc/hevc/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/dec/vp9/pipeline \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/vp/hal/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/vp/hal/platform_interface \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/enc/vp9/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/enc/av1/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/enc/vp9/packet \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/enc/hevc/features \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/dec/jpeg \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/hw/vdbox \
+    $(LOCAL_PATH)/../media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus/codec/hal/dec/mpeg2 \
+    $(LOCAL_PATH)/../media_softlet/linux/xe_lpm_plus_r0/encode/jpeg/ddi \
+    $(LOCAL_PATH)/../media_softlet/linux/xe_lpm_plus_r0/encode/hevc/ddi \
+    $(LOCAL_PATH)/../media_softlet/linux/xe_lpm_plus_r0/encode/vp9/ddi \
+    $(LOCAL_PATH)/../media_softlet/linux/xe_lpm_plus_r0/encode/av1/ddi \
+    $(LOCAL_PATH)/../media_softlet/linux/xe_lpm_plus_r0/ddi \
+    $(LOCAL_PATH)/../media_softlet/linux/xe_lpm_plus_r0/vp/ddi \
+    $(LOCAL_PATH)/../media_softlet/linux/xe_lpm_plus_r0/encode/avc/ddi \
+    $(LOCAL_PATH)/../media_softlet/agnostic/common/media_bin_mgr \
+    $(LOCAL_PATH)/../media_common/linux/common/cp/ddi \
+    $(LOCAL_PATH)/../media_common/linux/common/codec \
+    $(LOCAL_PATH)/media_softlet/agnostic/Xe_M/Xe_XPM_base/hw \
+    $(LOCAL_PATH)/../../libva/va \
+    $(LOCAL_PATH)/../cmrtlib/linux/hardware
 
 #LOCAL_CPP_FEATURES := rtti exceptions
 
diff --git a/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c b/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c
index 4f6e97798..5b6a72085 100644
--- a/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c
+++ b/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c
@@ -3864,6 +3864,7 @@ bool KernelDll_BuildKernel_CmFc(Kdll_State *pState, Kdll_SearchState *pSearchSta
     VP_RENDER_FUNCTION_ENTER;
 
     // Disable pop-up box window for STL assertion to avoid VM hang in auto test.
+#ifndef ANDROID
 #if (!LINUX)
     uint32_t prevErrorMode = ::SetErrorMode(SEM_FAILCRITICALERRORS | SEM_NOGPFAULTERRORBOX);
 #if defined(_MSC_VER)
@@ -3876,7 +3877,7 @@ bool KernelDll_BuildKernel_CmFc(Kdll_State *pState, Kdll_SearchState *pSearchSta
     _CrtSetReportFile(_CRT_ASSERT, _CRTDBG_FILE_STDERR);
 #endif
 #endif
-
+#endif
     pSearchState->KernelLink.dwSize  = DL_MAX_SYMBOLS;
     pSearchState->KernelLink.dwCount = 0;
     pSearchState->KernelLink.pLink   = pSearchState->LinkArray;
@@ -3994,7 +3995,7 @@ bool KernelDll_BuildKernel_CmFc(Kdll_State *pState, Kdll_SearchState *pSearchSta
     res = true;
 
 finish:
-#if (!LINUX)
+#if (!LINUX) && !defined(ANDROID)
     ::SetErrorMode(prevErrorMode);
 #endif
     return res;
diff --git a/media_softlet/linux/Xe_M_plus/ddi/media_libva_caps_mtl_base.cpp b/media_softlet/linux/Xe_M_plus/ddi/media_libva_caps_mtl_base.cpp
index 337dbbc89..b215e2799 100644
--- a/media_softlet/linux/Xe_M_plus/ddi/media_libva_caps_mtl_base.cpp
+++ b/media_softlet/linux/Xe_M_plus/ddi/media_libva_caps_mtl_base.cpp
@@ -37,6 +37,7 @@
 #include "media_ddi_decode_const_g12.h"
 #include "media_ddi_encode_const.h"
 #include "drm_fourcc.h"
+#include "va_android.h"
 
 const VAImageFormat m_supportedImageformatsXe_Lpm_Plus_Base[] =
 {   {VA_FOURCC_BGRA,           VA_LSB_FIRST,   32, 32, 0x0000ff00, 0x00ff0000, 0xff000000,  0x000000ff}, /* [31:0] B:G:R:A 8:8:8:8 little endian */
diff --git a/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp b/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
index 8765bf1ce..5fe17335f 100644
--- a/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
+++ b/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
@@ -2506,6 +2506,7 @@ void MosUtilities::MosTraceEvent(
                 MOS_FreeMemory(pTraceBuf);
             }
         }
+#ifndef ANDROID
 #if Backtrace_FOUND
         if (m_mosTraceFilter(TR_KEY_CALL_STACK))
         {
@@ -2526,6 +2527,7 @@ void MosUtilities::MosTraceEvent(
                 size_t ret = write(MosUtilitiesSpecificNext::m_mosTraceFd, traceBuf, nLen);
             }
         }
+#endif
 #endif
     }
     return;
-- 
2.17.1

